pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1550,32 +1638,47 @@ func doAppNetworkModifyAllUnderlayNetworks(
 	config types.AppNetworkConfig,
 	oldConfig types.AppNetworkConfig,
 	status *types.AppNetworkStatus,
-	ipsets []string) {
+	intfList []bool, ipsets []string) error {
 
 	for i := range config.UnderlayNetworkList {
 		log.Tracef("handleModify ulNum %d\n", i)
 		ulConfig := &config.UnderlayNetworkList[i]
 		oldulConfig := &oldConfig.UnderlayNetworkList[i]
 		ulStatus := &status.UnderlayNetworkList[i]
-		doAppNetworkModifyUnderlayNetwork(
-			ctx, status, ulConfig, oldulConfig, ulStatus, ipsets, false)
+		// the underlay network associationhave changed
+		if intfList[i] {
+			// the underlay interface has already been deactivated,
+			// activate it now
+			if err := appNetworkUNetActivate(ctx, config, status,
+				ulConfig, ulStatus, ulStatus.Vif, ulStatus.Mac,
+				ipsets); err != nil {
+				return err
```

Comment:
As commented elsewhere, why can't this be done in the code which detected the change and inactivated? Is there a reason you need to wait until you have potentially handled other network ID changes for other adapters or something??